### DISCUSSION TITLE
Resource placement

### DISCUSSION THREAD

**Discussion by dsogari** [liqotech/liqo#764] [Q&A] [answered]
The Kubernetes Cluster Federation project ([KubeFed](https://github.com/kubernetes-sigs/kubefed)) allows configuration of resource placement in federated clusters. This means, e.g., that I can choose in which clusters a workload resource should take place.

Liqo, on the other hand, provides "a unified and coherent view of your clusters". In that regard, I wonder if the placement behavior could be achieved by means of Kubernetes' [node affinity/taint](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) concepts?

For example, I could set a taint on all of a cluster's nodes like so: `kubectl taint node -l cluster=C1 dedicated=foo:NoSchedule`. Then, no application would be deployed on cluster C1, except if a corresponding toleration is applied to the pod template of the application deployment. Does this make sense?

**Comment by palexster** [comment, marked as answer]
Hi @dsogari, welcome! Yes, the taint/tolerations mechanism is completely compatible with Liqo, since the Liqo virtual node is seen as a normal node by the Kubernetes control plane.

**Reply by dsogari** [reply]
Hi @palexster, thanks! I joined your Slack workspace and I'm going to follow new developments through there.
